code/modules/clothing/head/_head.dm

```diff
@@ -15,6 +19,32 @@
 		var/mob/living/carbon/human/H = loc
 		H.update_hair()
 
+///Special throw_impact for hats to frisbee hats at people to place them on their heads/attempt to de-hat them.
+/obj/item/clothing/head/throw_impact(atom/hit_atom, datum/thrownthing/thrownthing)
+	. = ..()
+	///if the thrown object exists, has a target_zone from whatever threw it, isn't the thrower and the target zone isn't the head	
+	if(thrownthing && thrownthing.target_zone && thrownthing.thrower != hit_atom && thrownthing.target_zone != BODY_ZONE_HEAD)

```

Comment:
This is wrong. VERY wrong. Should be:
``` suggestion
	if(thrownthing.target_zone != BODY_ZONE_HEAD)
```
Why? Well basically, throwing can't ever make an item collide with the person throwing it. #44208
I don't see a reason why throw_impact would EVER get called without a throwdatum, and if it does, that's the thing that should get fixed.

Also try to use || instead of && in this kind of statement. This doesn't work at all, even if it were needed.